### PR TITLE
Fix EF based scaffolding

### LINKED ISSUE #481: Scaffolding should set ASPNETCORE_ENVIRONMENT=Development
The `DbContextOperations` API now ignores the `environment` parameter passed to it and expects 
`ASPNETCORE_ENVIRONMENT` variable to be set. 

https://github.com/aspnet/Scaffolding/blob/rel/2.0.0-preview1/src/Microsoft.VisualStudio.Web.CodeGeneration.EntityFrameworkCore/EntityFrameworkServices.cs#L448

cc @mlorbetske

### PR DESCRIPTION
**Pull request by prafullbhosale**
Fixes #481

### CODE REVIEW

File: src/Microsoft.VisualStudio.Web.CodeGeneration.EntityFrameworkCore/EntityFrameworkServices.cs [hunk 1]
```diff
@@ -441,12 +441,12 @@ private DbContext TryCreateContextUsingAppCode(Type dbContextType, Type startupT
...
                 var operationHandler = new OperationReportHandler();
                 var operationReporter = new OperationReporter(operationHandler);
+                // EF infers the environment (Development/ Production) based on the environment variable
+                // ASPNETCORE_ENVIRONMENT. This should already be set up by the CodeGeneration.Design process.
                 var dbContextOperations = new DbContextOperations(
                     operationReporter,
                     dbContextType.GetTypeInfo().Assembly,
-                    startupType.GetTypeInfo().Assembly,
-                    "Development",
-                    Path.GetDirectoryName(_projectContext.ProjectFullPath));
```
Comment (mlorbetske) [inline src/Microsoft.VisualStudio.Web.CodeGeneration.EntityFrameworkCore/EntityFrameworkServices.cs]: The startup type and path to project are no longer required?

File: src/Microsoft.VisualStudio.Web.CodeGeneration.EntityFrameworkCore/Templates/DbContext/NewDbContextFactory.cshtml [hunk 2]
```diff
@@ -21,6 +21,6 @@ namespace @Model.DbContextNamespace
     public class @factoryClassName : IDbContextFactory<@Model.DbContextTypeName>
     {
         public @Model.DbContextTypeName Create(string[] args) =>
-            @(Model.ProgramTypeName).BuildWebHost(args).Services.GetRequiredService<@Model.DbContextTypeName>();
+            @(Model.ProgramTypeName).BuildWebHost(args).Services.CreateScope().ServiceProvider.GetRequiredService<@Model.DbContextTypeName>();
```
Comment (mlorbetske) [inline src/Microsoft.VisualStudio.Web.CodeGeneration.EntityFrameworkCore/Templates/DbContext/NewDbContextFactory.cshtml]: I thought that these were not going to be required going forward

Comment (bricelam) [inline src/Microsoft.VisualStudio.Web.CodeGeneration.EntityFrameworkCore/Templates/DbContext/NewDbContextFactory.cshtml]: We also need to sort out if we even need an IDbContextFactory implementation.

File: src/Microsoft.VisualStudio.Web.CodeGeneration.EntityFrameworkCore/EntityFrameworkServices.cs (hunk 1 above)
Comment (bricelam) [inline src/Microsoft.VisualStudio.Web.CodeGeneration.EntityFrameworkCore/EntityFrameworkServices.cs]: We want to expose a public API for this in the 2.0.0 timeframe, so we isolate you from additional breaks.

Comment (prafullbhosale) [inline src/Microsoft.VisualStudio.Web.CodeGeneration.EntityFrameworkCore/EntityFrameworkServices.cs]: @mlorbetske, the startupType's assembly is still needed. 
The `environment` and `contentRootPath` parameters are no longer needed. 
See https://github.com/aspnet/EntityFramework/issues/7050#issuecomment-297833793 for reference.

File: src/Microsoft.VisualStudio.Web.CodeGeneration.EntityFrameworkCore/Templates/DbContext/NewDbContextFactory.cshtml (hunk 2 above)
Comment (prafullbhosale) [inline src/Microsoft.VisualStudio.Web.CodeGeneration.EntityFrameworkCore/Templates/DbContext/NewDbContextFactory.cshtml]: Created https://github.com/aspnet/Scaffolding/issues/485. 
Will need to keep this for now so the tests don't break.

File: src/Microsoft.VisualStudio.Web.CodeGeneration.EntityFrameworkCore/EntityFrameworkServices.cs (hunk 1 above)
Comment (mlorbetske) [inline src/Microsoft.VisualStudio.Web.CodeGeneration.EntityFrameworkCore/EntityFrameworkServices.cs]: Ah, missed the add for startup type, only saw the delete, sorry

**Pull request merged**